### PR TITLE
remove vllm-check/tgi-check init-container

### PR DESCRIPTION
**Pull request by robinliubin**
add wget vllm into h2ogpt container

fixing: https://github.com/h2oai/h2ogpt/issues/1603

### CODE REVIEW

File: helm/h2ogpt-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -190,10 +158,29 @@ spec:
...
+          {{- if and .Values.tgi.enabled }}
+          args:
+          - >
+            until wget -O- http://{{ include "h2ogpt.fullname" . }}-tgi-inference:{{ .Values.tgi.service.port }}/ >/dev/null 2>&1;
+              do
+                echo "Waiting for inference service to become ready...";
+                sleep 5;
+              done
+              
+            python3 /workspace/generate.py
```
[the remark this answers is not in the document]
Comment (robinliubin) [reply helm/h2ogpt-chart/templates/deployment.yaml:181]: @joby-h20 is testing this, will update if needed.
currently refering to the h2ogpt.stack enabled option:
https://github.com/h2oai/h2ogpt/blob/c3561c2777564c8ee1fe22f01e940871832c9ace/helm/h2ogpt-chart/templates/deployment.yaml#L181-L192

[the remark this answers is not in the document]
Comment (achraf-mer) [reply helm/h2ogpt-chart/templates/deployment.yaml:181]: also, why `and` in:
```
{{- if and .Values.tgi.enabled }}
```
?

[the remark this answers is not in the document]
Comment (robinliubin) [reply helm/h2ogpt-chart/templates/deployment.yaml:181]: typo, fixed

**Comment by joby-h20** [comment]
Tested by injecting the changes directly to deployment manifest.

https://h2oai.slack.com/archives/C06R0QAP54N/p1715188293152829?thread_ts=1715178088.028339&cid=C06R0QAP54N

**Review by joby-h20 — APPROVED** [review]
lgtm

**Review by achraf-mer — APPROVED** [review]
added one comment about using a list or arguments, but if tested and works fine then LGTM, thanks.

**Review by EshamAaqib — CHANGES_REQUESTED** [review]
Should be fine since  it checks for `{{- if .Values.vllm.enabled }}` . Didn't see it before :)

**Review by EshamAaqib — APPROVED** [review]
LGTM

**Comment by ozahavi** [comment]
This change **may** cause issues if we ever want to introduce liveness\readiness probes to h2oGPT.

**Comment by EshamAaqib** [comment]
If the inference service is vLLM and if Model Lock is used the check will not be needed

**Comment by robinliubin** [comment]
> This change **may** cause issues if we ever want to introduce liveness\readiness probes to h2oGPT.

thanks @ozahavi, would you elaborate more about moving wget into h2ogpt container would be an issue for liveness/readiness?

**Comment by robinliubin** [comment]
> ^ +1, If the inference service is vLLM and if Model Lock is used the check will not be needed, it can be disabled completely
like this:
```yaml
{{- if and .Values.vllm.enabled (not .Values.h2ogpt.externalLLM.enabled) }}
```

**Comment by EshamAaqib** [comment]
> > ^ +1, If the inference service is vLLM and if Model Lock is used the check will not be needed, it can be disabled completely
> 
> 
> 
> like this:
> 
> ```yaml
> 
> {{- if and .Values.vllm.enabled (not .Values.h2ogpt.externalLLM.enabled) }}
> 
> ```

Yes, and also you can stick with Model Lock itself for locally hosted LLMs and if the inference is vLLM the wget check is not needed (if you use Model Lock)

**Comment by robinliubin** [comment]
> > > ^ +1, If the inference service is vLLM and if Model Lock is used the check will not be needed, it can be disabled completely
> > 
> > 
> > like this:
> > ```yaml
> > {{- if and .Values.vllm.enabled (not .Values.h2ogpt.externalLLM.enabled) }}
> > ```
> 
> Yes, and also you can stick with Model Lock itself for locally hosted LLMs and if the inference is vLLM the wget check is not needed (if you use Model Lock)
> 
> Ex: https://github.com/h2oai/public-cloud-infrastructure/blob/develop/hamc/hadc-customer-account/terraform/environment/applications/main-h2ogpt.tf#L5

like this: 
```yaml
{{- if and .Values.vllm.enabled (not .Values.h2ogpt.externalLLM.modelLock) }}
```

**Comment by robinliubin** [comment]
tested with: `helm template h2oai helm/h2ogpt-chart --set vllm.enabled=true --set h2ogpt.externalLLM.modelLocak=null`
it rendered to:
```
       args:
          - >
            until wget -O- http://h2oai-h2ogpt-vllm-inference:5000/v1/models >/dev/null 2>&1;
              do
                echo "Waiting for inference service to become ready...";
                sleep 5;
              done
              
            python3 /workspace/generate.py
```

**Comment by EshamAaqib** [comment]
> > > > ^ +1, If the inference service is vLLM and if Model Lock is used the check will not be needed, it can be disabled completely
> 
> > > 
> 
> > > 
> 
> > > like this:
> 
> > > ```yaml
> 
> > > {{- if and .Values.vllm.enabled (not .Values.h2ogpt.externalLLM.enabled) }}
> 
> > > ```
> 
> > 
> 
> > Yes, and also you can stick with Model Lock itself for locally hosted LLMs and if the inference is vLLM the wget check is not needed (if you use Model Lock)
> 
> > 
> 
> > Ex: https://github.com/h2oai/public-cloud-infrastructure/blob/develop/hamc/hadc-customer-account/terraform/environment/applications/main-h2ogpt.tf#L5
> 
> 
> 
> like this: 
> 
> ```yaml
> 
> {{- if and .Values.vllm.enabled (not .Values.h2ogpt.externalLLM.modelLock) }}
> 
> ```

Yup lgtm

**Pull request merged**